pyproject.toml

```diff
@@ -113,6 +113,37 @@ include = [
     "/cosmos",
 ]
 
+######################################
+# TESTING
+######################################
+
+[tool.hatch.envs.tests]
+dependencies = [
+    "pytest>=6.0",
+    "pytest-split",
+    "pytest-dotenv",
+    "requests-mock",
+    "pytest-cov",
+    "pytest-describe",
+    "types-requests",
+    "mypy",
+    "sqlalchemy-stubs", # Change when sqlalchemy is upgraded https://docs.sqlalchemy.org/en/14/orm/extensions/mypy.html
+
+    "apache-airflow-providers-docker>=3.5.0",
+    "apache-airflow-providers-cncf-kubernetes>=5.1.1",
+]
+
+[[tool.hatch.envs.tests.matrix]]
+python = ["3.7", "3.8", "3.9", "3.10"]
+
+[tool.hatch.envs.tests.scripts]
+test = "pytest ."
+test-cov = "pytest --cov=cosmos --cov-report=term-missing --cov-report=xml ."
```

Comment:
One of the benefits of having separate directories for tests and code is that we could check the coverage only in the actual non-test modules. Currently, we're checking the coverage both in tests and code, which can be very misleading.

Reply:
I just fixed this issue - good catch!